### PR TITLE
Always use up to date callback if no dependencies are passed

### PR DESCRIPTION
**Pull request by tbezman**
### [Reproduction](https://codesandbox.io/s/interesting-swirles-cqb8o9?file=/src/App.tsx)

In the sandbox, you can see we're never getting past 1 because the callback doesn't get updated.

### The Dependencies Array

- Adding a deps array I don't think is the correct API.
- What we really want is something like the [`useEvent` hook](https://github.com/streamich/react-use/blob/master/docs/useEvent.md) but that probably wont' be around for a while.
- The workaround is to always store the latest callback in a ref (similar to how useEvent was intended to work).

### Breaking Changes
In order not to break people that were using the dependencies array, the code got a little bit uglier. We need a mechanism to determine whether or not the caller actually intended to use the deps array or not.

I added some tests to ensure all is compatible. This API is much more compatible with how people expect the library to work without trading off performance problems

### CODE REVIEW

File: src/useHotkeys.ts [hunk 1]
```diff
@@ -32,9 +32,17 @@ export default function useHotkeys<T extends HTMLElement>(
...
-  const _deps: DependencyList = options instanceof Array ? options : dependencies instanceof Array ? dependencies : []
+  const _deps: DependencyList | undefined = options instanceof Array ? options : dependencies instanceof Array ? dependencies : undefined
+
+  const memoisedCB = useCallback(callback, _deps ?? [])
+  const cbRef = useRef<HotkeyCallback>(memoisedCB);
+
+  if(_deps) {
+    cbRef.current = memoisedCB;
+  } else {
+   cbRef.current = callback;
```
Comment (Robinnnnn) [inline src/useHotkeys.ts:43]: did this tab not get linted by prettier?

**Review by Robinnnnn — APPROVED** [review]

File: src/useHotkeys.ts (hunk 1 above)
Comment (tbezman) [reply src/useHotkeys.ts:43]: oof

Comment (tbezman) [reply src/useHotkeys.ts]: looks like running prettier reformats a bunch of stuff, i'll just fix the one line

File: tests/useHotkeys.test.tsx [hunk 2]
```diff
@@ -1080,3 +1080,59 @@ test('Should ignore modifiers if option is set', async () => {
 
   expect(callback).toHaveBeenCalledTimes(2)
 })
+
+
+
+describe.only('dependencies array', function() {
```
Comment (JohannesKlauss) [inline tests/useHotkeys.test.tsx:1086]: Please remove the describe wrapper

**Review by JohannesKlauss — CHANGES_REQUESTED** [review]

**Comment by JohannesKlauss** [comment]
Thank you for the work. But I don't see how this improves anything.
With your PR users are able to pass in their own function returned by `useCallback`, but why would they want to do that when you can just inline your function and declare their dependencies? You are pulling out functionality of the hook and move it to user land. But maybe I am missing the point here?

**Comment by Robinnnnn** [comment]
@JohannesKlauss as a user i'd be confused by the behavior exhibited in the [codesandbox](https://codesandbox.io/s/interesting-swirles-cqb8o9?file=/src/App.tsx)

**Comment by JohannesKlauss** [comment]
Ah ok, I do see the point in that.

**Review by Robinnnnn — APPROVED** [review]

**Comment by JohannesKlauss** [comment]
Thank you very much, this landed in 4.3.5

**Comment by LuanScudeler** [comment]
@tbezman , @JohannesKlauss I've found some inconsistency between the live examples and the content of the documentation in this page https://react-hotkeys-hook.vercel.app/docs/documentation/useHotkeys/setting-callback-dependencies  and then end up here.

So for clarification, the decision is to support both usages? one being memoization handled by the hook in case we provide the dependency array param, and second is memoization handled by the user when we don't pass a dependency array.

**Pull request merged**